lightning/src/chain/channelmonitor.rs

```diff
@@ -3111,10 +3111,21 @@ impl<Signer: Sign> ChannelMonitorImpl<Signer> {
 		F::Target: FeeEstimator,
 		L::Target: Logger,
 	{
-		self.onchain_events_awaiting_threshold_conf.retain(|ref entry| if entry.txid == *txid {
-			log_info!(logger, "Removing onchain event with txid {}", txid);
-			false
-		} else { true });
+		let mut height = None;
+		for entry in self.onchain_events_awaiting_threshold_conf.iter() {
+			if entry.txid == *txid {
+				height = Some(entry.height);
+				break;
+			}
+		}
+
+		if let Some(height) = height {
+			log_info!(logger, "transaction_unconfirmed of txid {} implies height {} was reorg'd out", txid, height);
+			self.onchain_events_awaiting_threshold_conf.retain(|ref entry| entry.height < height);
```

Comment:
Okay I think it's correct to not call `ChannelMonitor::block_disconnected` as `chain::Confirm` doc implies to call `best_block_updated`. This method updates our `best_block` and call `OnchainTxHandler::block_disconnected`.